pkg/controller/awsprivatelink/cleanup.go

```diff
@@ -171,6 +176,25 @@ func (r *ReconcileAWSPrivateLink) cleanupHostedZone(awsClient awsclient.Client,
 	}
 
 	hzLog := logger.WithField("hostedZoneID", hzID)
+
+	// Determine if hzID is one of SharedHostedZoneDomains
+	var isSharedHostedZone bool
+	zoneResp, err := awsClient.GetHostedZone(&route53.GetHostedZoneInput{
+		Id: aws.String(hzID),
+	})
+	if awsErrCodeEquals(err, "NoSuchHostedZone") {
+		return nil // no more work
+	}
+	if err != nil {
+		hzLog.WithError(err).Error("failed to get the hosted zone")
+		return err
+	}
+	for _, domain := range r.controllerconfig.SharedHostedZoneDomains {
+		if strings.EqualFold(domain, strings.TrimSuffix(aws.StringValue(zoneResp.HostedZone.Name), ".")) {
+			isSharedHostedZone = true
+		}
+	}
```

Comment:
This could probably reuse `selectDomain()`, something like:

```go
isSharedHostedZone := selectDomain(apiDomain, r.controllerconfig.SharedHostedZoneDomains) == zoneResp.HostedZone.Name
```